apps/common/components/Table.tsx

```diff
@@ -0,0 +1,97 @@
+import {useCallback, useMemo, useState} from 'react';
+import {sort} from '@veYFI/utils';
+import IconChevronPlain from '@common/icons/IconChevronPlain';
+
+import type {ReactElement} from 'react';
+
+type TSortOrder = 'asc' | 'desc' | '';
```

Comment:
nit: I think here we could go with `null` as it's more aligned with what the order is, wdyt?

```suggestion
type TSortOrder = 'asc' | 'desc' | null;
```

Reply:
I'm not sure that's what we want, because we do want to have a `null`.

A column that is not sorted should just have sort order as `null`, the reason is that otherwise the little arrows on top depend on that knowledge (3 they are either active asc/desc or disabled). Basically, when the table loads 1 column could be sorted (say APY desc), but all the other columns should have sorting disabled (if sortable), if not sortable then they don't have the sorting caret.

Wdyt? Or am I missing something?

Reply:
> how I see it is only two sorting orders exist, either ascending or descending

There's a 3rd one, unsorted. I believe that must have been the idea with the empty string.

> Either we have a sortedBy column active or not.

Hmmm sorry to be pain 😅 but we could sort multiple columns. Imagine the following scenario. I sort A first `asc`, then B `asc` and finally C `asc`, so the table

```
| A | B | C |
|---|---|---|
| 2 | 2 | 3 |
| 1 | 1 | 2 |
| 2 | 1 | 1 |
```
becomes
```
| A | B | C |
|---|---|---|
| 1 | 1 | 1 |
| 1 | 1 | 2 |
| 2 | 2 | 3 |
```

It's a hypothetical scenario, but maybe there will be tables where I'd like to sort by more than one column.

Reply:
> if we need the multiple column sort, we can implement it at the time we need, wdyt?

agreed, YAGNI principle 👍